Source/Dafny/Compiler-js.cs

```diff
@@ -2183,19 +2183,17 @@ protected class ClassWriter : IClassWriter {
       wr.Write(")");
     }
 
-    protected override void EmitCollectionBuilder_New(CollectionType ct, Bpl.IToken tok, TargetWriter wr) {
-      if (ct is SetType) {
-        wr.Write("new _dafny.Set()");
-      } else if (ct is MultiSetType) {
-        wr.Write("new _dafny.MultiSet()");
-      } else if (ct is MapType) {
-        wr.Write("new _dafny.Map()");
-      } else {
-        Contract.Assume(false);  // unepxected collection type
-      }
+    protected override void EmitSetBuilder_New(TargetWriter wr, SetComprehension e, string collectionName) {
+      var wrVarInit = DeclareLocalVar(collectionName, null, null, wr);
+      wrVarInit.Write("new _dafny.Set()");
```

Comment:
Compiler-js defines constants like DafnySetClass to hold the string "_dafny.Set", as do the other compiler files. These constant strings should be used consistently or not at all -- here and in many other places

Reply:
Done for the collection types (except in the Go compiler, which uses each of those names only once).